DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs

```diff
@@ -19,48 +21,92 @@
     public class ReportsController : Controller
     {
         private readonly IActivityService activityService;
+        private readonly ICourseCategoriesDataService courseCategoriesDataService;
+        private readonly ICourseDataService courseDataService;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserDataService userDataService;
 
-        public ReportsController(IActivityService activityService)
+        public ReportsController(
+            IActivityService activityService,
+            IJobGroupsDataService jobGroupsDataService,
+            IUserDataService userDataService,
+            ICourseDataService courseDataService,
+            ICourseCategoriesDataService courseCategoriesDataService
+        )
         {
             this.activityService = activityService;
+            this.jobGroupsDataService = jobGroupsDataService;
+            this.userDataService = userDataService;
+            this.courseDataService = courseDataService;
+            this.courseCategoriesDataService = courseCategoriesDataService;
         }
 
         public IActionResult Index()
         {
             var centreId = User.GetCentreId();
+            var adminId = User.GetAdminId()!.Value;
+            var adminUser = userDataService.GetAdminUserById(adminId)!;
+
+            int? courseCategoryId = null;
+            var validatedCategoryId = adminUser.CategoryId == 0 ? courseCategoryId : adminUser.CategoryId;

```

Comment:
Why do we define the courseCategoryId above? Also, there's no "validation" of category id happening here - maybe "categoryIdFilter" would be a better name?

Reply:
Odd holdover from faking the cookie with query strings, not sure why I left it that way...